bitcoin/shadouble.c

```diff
@@ -1,16 +1,21 @@
 #include "shadouble.h"
+#include <bitcoin/groestl.h>
 #include <ccan/mem/mem.h>
 #include <common/type_to_string.h>
+#include <common/utils.h>
 
 void sha256_double(struct sha256_double *shadouble, const void *p, size_t len)
 {
 	sha256(&shadouble->sha, memcheck(p, len), len);
-	sha256(&shadouble->sha, &shadouble->sha, sizeof(shadouble->sha));
+	groestlhash((void *)&shadouble->sha, (void *)p, len);
 }
 
 void sha256_double_done(struct sha256_ctx *shactx, struct sha256_double *res)
 {
 	sha256_done(shactx, &res->sha);
-	sha256(&res->sha, &res->sha, sizeof(res->sha));
+#ifdef COMPILE_FOR_BITCOIN
```

Comment:
We can drop statically by macro COMPILE_FOR_BITCOIN since we are plain c-grs-lightning